src/Microsoft.Health.Fhir.Shared.Client/FhirClient.cs

```diff
@@ -493,6 +493,22 @@ public async Task<FhirResponse<Bundle>> PostBundleAsync(Resource bundle, Cancell
             return await CreateResponseAsync<Bundle>(response);
         }
 
+        public async Task<FhirResponse<Bundle>> PostBundleAsyncWithValidationHeader(Resource bundle, string profileValidation, CancellationToken cancellationToken = default)
```

Comment:
Can this instead be taken as bool arg and then converted to string internally?

Reply:
This is used only in E2E tests where profileValidation is being passed as a string value. Is there any advantage of passing it as boolean? Having it as boolean type needs it to convert to boolean in the test method from where this method is being called and convert back into a string in this method.

Reply:
Given this api is now going to be exposed to consumers using this client, IMO expecting them to pass a string literal instead of a bool is not ideal unless there are strong reasons to have the arg be string. It is more intuitive to pass a bool instead here as this is after all a flag.